src/hooks/usePrimaryToolbarFilterConfigWrapper.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, Dropdown, DropdownItem, DropdownToggle, ToolbarExpandIconWrapper, TreeView, TreeViewDataItem } from "@patternfly/react-core";
+import { AngleDownIcon } from "@patternfly/react-icons";
+import { ColumnsMetada, usePrimaryToolbarFilterConfig } from "@redhat-cloud-services/insights-common-typescript";
+import React from "react";
+import { useEffect, useMemo } from "react"
+import { ClearEventLogFilters, EventLogFilterColumn, EventLogFilters, SetEventLogFilters } from "../components/Notifications/EventLog/EventLogFilter";
+import { Schemas } from "../generated/OpenapiNotifications";
+
+interface TreeFilterProps {
+    groups: readonly Schemas.Facet[],
+    items: readonly Schemas.Facet[],
+    placeholder: string
+}
+
+export const TreeDropdownFilter: React.FunctionComponent<TreeFilterProps> = (props) => {
+    const [isFocused, setIsFocused] = React.useState(true)
+    const data: TreeViewDataItem[] = props.groups.map(group => ({
+        id: group.name,
+        name: group.displayName,
+        checkProps: { checked: false },
+        children: props.items.map(item => ({
+            id: item.name,
+            name: item.displayName,
+            checkProps: { checked: false }
+        }))
+
+    }))
+
+    console.log("render")
+    return (
+        <Dropdown
+            onSelect={() => setIsFocused(!isFocused)}
+            toggle={
+                <DropdownToggle toggleIndicator={AngleDownIcon}>{props.placeholder}</DropdownToggle>
```

Comment:
This is the toggle (the visible part of the Dropdown when is closed).
Here is where you want to control the behavior of the opening/closing.

```suggestion
                <DropdownToggle onToggle={ () => setIsFocused(prev => !prev) } toggleIndicator={AngleDownIcon}>{props.placeholder}</DropdownToggle>
```

Reply:
Thank you, that fixes my issue... pretty careless mistake of me. I totally forgot about the component used specifically for toggling the dropdown.

Reply:
Nothing to worry, It happens all the time - an extra pair of eyes is always helpful.